src/client.rs

```diff
@@ -271,41 +271,28 @@ impl Client {
         addr: (std::net::IpAddr, u16),
         url: &Url,
     ) -> Result<Stream, ClientError> {
-        // TODO: Allow the connect timeout to be configured
-        let timeout_duration = tokio::time::Duration::from_secs(5);
-
```

Comment:
Are we sure that a connection timeout is what is causing the crash?

It might be handling of errors after the timeouts that cause the crash or some of the other changes here.

I'd really suggest keeping the timeout if possible.